hip/stop/batch_criteria.hip.hpp

```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_HIP_STOP_BATCH_CRITERIA_HIP_HPP_
+#define GKO_HIP_STOP_BATCH_CRITERIA_HIP_HPP_
+
+
+#include <ginkgo/core/base/math.hpp>
+
+
+namespace gko {
+namespace kernels {
+namespace hip {
```

Comment:
why can't this be in cuda_hip?

Reply:
The whole batch ecosystem has an extremely complex include structure, the untangling of which would make this PR much too complicated. I started with it and realized that it would be too much effort, so I reverted all changes related to batch